### DISCUSSION TITLE
Using Environment Variable in .env file to run a test container

### DISCUSSION THREAD

**Discussion by RyanKellyReply** [danny-avila/LibreChat#3308] [Q&A]
Hi, 
Perhaps a very basic question, my programming skills are basic. I am trying to deploy a test container for Librechat using a env variable in my .env file where i expose the open ai API key at runtime rather than exposing my API key itself in the codebase. 

In the .env file I am first declaring the environment variable with an empty value  OPEN_AI_KEY=

Later on in the .env file, I am setting the value where i normally insert my open ai key as such OPENAI_API_KEY=${OPEN_AI_KEY}  which sets it to the value of the environment variable above where i will pass the actual value at runtime

When I go to run the test container in desktop I am declaring the variable OPEN_AI_KEY= myvalue but I am getting the following error TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined 

so its expecting a string but getting undefined which must mean i am not declaring it properly. Do i need to change any other files to get this working such as the crpyto.js file? 

Any help or best practice would be much appreciated

**Comment by danny-avila** [comment]
Try copying `.env.example` as `.env` first before setting any variables and see if that works for you

**Reply by RyanKellyReply** [reply]
and remove the user provided and declare the key at runtime ?

**Reply by RyanKellyReply** [reply]
ive got it running locally in a seperate container where I have declared my key in the file
